app/models/carto_uploader.rb

```diff
@@ -1,28 +1,44 @@
+require 'csv'
 require 'uri'
 require 'net/http'
+require 'json'
 
 class CartoUploader
+  #--------------------------------
   CARTODB_ACCOUNT = ENV["CARTODB_ACCOUNT"]
   CARTODB_APIKEY = ENV["CARTODB_APIKEY"]
   CARTODB_URI_GET = "https://#{CARTODB_ACCOUNT}.cartodb.com/api/v2/sql?q=__SQLQUERY__&api_key=#{CARTODB_APIKEY}"
   CARTODB_URI_POST = "https://#{CARTODB_ACCOUNT}.cartodb.com/api/v2/sql"
-  CARTODB_TABLE = ENV["CARTODB_TABLE"]
-  
-  CARTODB_INSERT_MODE_INDIVIDUAL = "ind"
-  CARTODB_INSERT_MODE_MASS = "mass"
+  CARTODB_TABLE = ENV["CARTODB_TABLE"]  
+  CLASSIFICATIONS_PER_BATCH = 20
+  #--------------------------------
 
-  def initialize(mode = CARTODB_INSERT_MODE_MASS)
-    @mode = mode
+  def initialize()
   end
   
   def upload(data)
-    allVals = []  #For Mass Insertion mode only.
-    keys = []     #Declaration for Mass Insertion mode.
+    #Determine what's already in the database
+    #--------------------------------
+    latestClassificationID = getLatestClassificationID()
+    expectedInsertions = 0
```

Comment:
Thoughts: What _should_ happen when `latestClassificationID` is `0`?

Option A: Burn it all then start anew
```
if (latestClassificationID == 0)
  begin
    tryToTruncateTable()
  rescue
    raise StandardError, "Even TRUNCATE-ing the table didn't work, maybe CartoDB is unavailable dude"
  end
end
```

Option B: Roll over and die
```
if (latestClassificationID == 0)
  raise StandardError, "Could not contact CartoDB database to determine latest Classification ID. Help!"
end
```